src/cpu.rs

```diff
@@ -27,31 +37,48 @@ fn parse(contents: &str) -> HashMap<&str, &str> {
         .collect()
 }
 
-pub fn get_cpu_model() -> Option<String> {
-    let contents = match read_to_string(CPU_INFO_PATH) {
-        Ok(c) => c,
-        Err(e) => {
-            eprintln!("Failed to read {}, {}", CPU_INFO_PATH, e);
-            return None;
-        }
-    };
+/// Gets cpu model name and core count and returns result as tuple
+fn get_cpu() -> (String, u32) {
+    let contents = read_to_string(CPU_INFO_PATH).expect("Failed to read /proc/cpuinfo");
     let cpu_info = parse(&contents);
-    Some(cpu_info.get("model name")?.to_string())
+    let model = cpu_info
+        .get("model name")
+        .expect("Failed to parse CPU model")
+        // Remove some unnecessary patterns from model name
+        .replace("(TM)", "")
+        .replace("(R)", "")
+        .replace(" Core", "")
+        .replace(" CPU", "")
+        .replace(" Processor", "")
```

Comment:
hmm... how does final output look for you?

Reply:
this is what i get: `CPU: AMD Ryzen 5 2600 Six-Core (6)`

Reply:
> hmm... what does the final model look like for you?

Now looks like this: `Intel i5-8400 @ 2.80GHz`. Previously looked like this: `Intel(R) Core(TM) i5-8400 CPU @ 2.80GHz`

Reply:
How does it look like for you?

Reply:
> this is what i get: `CPU: AMD Ryzen 5 2600 Six-Core (6)`

Hmm, maybe that 'Six-Core' part is a bit unnecessary. Neofetch has a long [list](https://github.com/dylanaraps/neofetch/blob/master/neofetch#L2345-#L2366) of patterns that they remove from the model name....

Reply:
maybe we can have the same

Reply:
Then we would probably have to use the regex library, which would add another dependency but maybe that's ok 😄

Reply:
i was thinking more of having a list with patterns defined and then just loop through them and replace with empty strings (for now)